test/unit/collector/facts.test.js

```diff
@@ -829,7 +829,12 @@ test('host facts', async (t) => {
 
     ctx.nr.agent = helper.loadMockedAgent(structuredClone(DISABLE_ALL_DETECTIONS))
     ctx.nr.agent.config.utilization = null
-    ctx.nr.agent.config.getHostnameSafe = () => 'localhost'
+    ctx.nr.agent.config.getHostnameSafe = (gcpId) => {
```

Comment:
You should remove this mock and fallback to the actual logic. These could also just move to the config unit tests